Pax.cs

```diff
@@ -300,6 +304,24 @@ private static IPacketProcessor InstantiatePacketProcessor(Type type)
       IPacketProcessor pp = PacketProcessorHelper.InstantiatePacketProcessor(type, arguments);
       if (pp == null)
         Console.WriteLine("Couldn't instantiate {0}", type.FullName);
+
+      // FIXME could extract this into a separate function to check version
+      //       compatibility, as part of a more general extraction of an API
+      //       from the Frotend class.
+      System.Version version = new System.Version(Frontend.Version);
+      if (pp is IVersioned) {
+        IVersioned pp_v = pp as IVersioned;
+        if (pp_v.expected_major_Pax_version() != version.Major ||
+            pp_v.expected_minor_Pax_version() != version.Minor) {
```

Comment:
Thanks very much for the feedback, and I agree with your proposal. I'll also add a note specifying that we'll use semantic versioning.